components/SelectField.tsx

```diff
@@ -0,0 +1,150 @@
+import { FC, useState, useEffect, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  HTMLChakraProps,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  useDisclosure,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { TriangleDownIcon } from '@chakra-ui/icons'
+
+type OptionType = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any
+  label: string | ReactNode
+  helperText: string | ReactNode
+}
+
+interface OptionProps extends HTMLChakraProps<'div'> {
+  option: OptionType
+}
+
+const Option: FC<OptionProps> = ({ option, ...props }) => (
+  <Box {...props}>
+    <Box className="select-field_option_label">{option?.label}</Box>
+    <Box className="select-field_option_text">{option?.helperText}</Box>
+  </Box>
+)
+
+interface SelectFieldProps extends FlexProps {
+  label: string
+  value?: OptionType
+  options?: OptionType[]
+  renderOption?: (option: OptionType) => ReactNode
+  renderSelected?: (option: OptionType) => ReactNode
+  onSelectOption?: (option: OptionType) => void
+}
+
+const SelectField: FC<SelectFieldProps> = ({
+  label,
+  value,
+  options = [],
+  renderOption,
+  renderSelected,
+  onSelectOption,
+  ...props
+}) => {
+  const [val, setVal] = useState<OptionType | null>(null)
+  const { onOpen, onClose, isOpen } = useDisclosure()
+  const styles = useMultiStyleConfig('SelectField', props)
+
+  useEffect(() => {
+    setVal(value || null)
```

Comment:
Minor, but probably would make more sense to default `value = null` on line 47, and then this line can just always be `setVal(value)`?